enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,259 @@
+
+1. **Lightweight Solution:** Deploying and configuring Prometheus Adapter requires additional effort compared to Metrics Server. Prometheus Adapter is tightly coupled to Prometheus and this creates a bottleneck for allowing monitoring as optional feature. See [MON-3152](https://issues.redhat.com/browse/MON-3152).
+
+1. **Better Accuracy:**  By default, the Metrics Server takes into account the timestamps of metric samples, allowing for accurate tracking of resource utilization over time. This means you can rely on the Metrics Server to provide a more precise understanding of how resources are being utilized by your applications.
+
+1. **Scability:** Starting from `v0.5.0` Metrics Server comes with default resource requests that guarantee good performance for most cluster configurations and it continues to do [scale testing](https://github.com/kubernetes-sigs/metrics-server/blob/077a462cd660beef705afb0a4a14a707ef39c4f2/FAQ.md#how-large-can-clusters-be) which is not the case for Prometheus Adapter.
+
+### User Stories
+
+#### Story 1
+
+* In order to provide [optional built-in monitoring](https://issues.redhat.com/browse/MON-3152) feature we need to provide a way to support HPA and VPA without a Prometheus-based solution.
+
+#### Story 2
+
+* As a cluster admin, I seek a lightweight and efficient solution with improved accuracy for monitoring resource metrics within my cluster.
+
+#### Story 3
+
+* As a cluster admin, I want a simplified monitoring architecture in OpenShift, so that I can reduce the complexity and maintenance overhead of the monitoring system.
+
+### Goals
+
+* Replace Prometheus Adapter with Metrics Server while maintaining feature parity with the existing stack 
+* Ensure no performance regression occurs when replacing Prometheus Adapter with Metrics Server
+* Improved scale up time for autoscaling
+
+### Non-Goals
+
+* Allow optional cluster monitoring stack
+* Moving `openshift-monitoring/cluster-monitoring-config` config-map to a Custom Resource
+
+## Proposal
+
+The proposal is to replace the Prometheus Adapter with the Metrics Server for monitoring resource metrics in OpenShift.
+
+### Workflow Description
+
+The workflow for using the Metrics Server in OpenShift for monitoring resource metrics would involve the following steps:
+
+1. User enables `TechPreviewNoUpgrade` featureset
+
+    ```yaml
+    apiVersion: config.openshift.io/v1
+    kind: FeatureGate
+    metadata:
+      name: cluster
+    spec:
+      featureSet: TechPreviewNoUpgrade
+    ```
+
+    With this configuration, the cluster monitoring operator should:
+    1. Deploy the `metrics-server` resources in the `openshift-monitoring` namespace
+    1. Verify that the `metrics-server` service is functional by querying the `/readyz` endpoint
```

Comment:
I think that we gave up the idea to probe the /readyz endpoint but rather leverage the Kubernetes capabilities:
```suggestion
    1. Wait for the metrics-server task to be complete which involves the metrics-server deployment and service being ready.
```